src/box/box.cc

```diff
@@ -2246,8 +2246,38 @@ box_set_bootstrap_leader_record(void)
 }
 
 int
-box_make_bootstrap_leader(void)
+box_make_bootstrap_leader(bool graceful)
 {
+	/* Bootstrap strategy is read by the time instance uuid is known. */
```

Comment:
I think you should also reset `is_graceful_supervised_bootstrap_requested` and `is_supervised_bootstrap_leader` flags in `box_set_bootstrap_strategy()`, if strategy happens to be not 'supervised'.

Reply:
I think it worth to do. It doesn't affect usual scenarios, where the strategy is not changed. However, it brings more expected behavior in some unusual ones, where the strategy is changed. Thanks for the idea!

Reply:
Added in the corresponding commits.

Reply:
Thanks!